src/lib/pmd/PmdSupport.ts

```diff
@@ -50,14 +56,33 @@ export abstract class PmdSupport {
     });
   }
 
-  protected abstract buildCommandArray(): [string, string[]];
+  protected abstract buildCommandArray(): Promise<[string, string[]]>;
 
   protected async runCommand(): Promise<[boolean,string]> {
-    const [command, args] = this.buildCommandArray();
+    const [command, args] = await this.buildCommandArray();
 
     return new Promise<[boolean,string]>((res, rej) => {
       const cp = child_process.spawn(command, args);
       this.monitorChildProcess(cp, res, rej);
     });
   }
+
+  protected async getCustomClasspath(): Promise<string[]> {
```

Comment:
Building classpath from all the custom paths provided for each language under PMD.